### PR TITLE
Remove unnecessary ruby filters

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
### What was the end-user problem that led to this PR?

The problem was there's a lot of unnecessary code, specially in specs, that's never run.

### What was your diagnosis of the problem?

My diagnosis was that since dropping support for old ruby versions, this code is no longer necessary.

### What is your fix for the problem, implemented in this PR?

My fix is to remove the code.

### Why did you choose this fix out of the possible options?

I chose this fix because red is great on diffs.

### CODE REVIEW

File: spec/other/compatibility_guard_spec.rb [hunk 1]
```diff
@@ -4,15 +4,6 @@
   context "when the bundler version is 2+" do
     before { simulate_bundler_version "2.0.a" }
 
-    context "when running on Ruby < 2.3", :ruby => "< 2.3" do
```
Comment (simi) [inline spec/other/compatibility_guard_spec.rb]: Any chance to keep this spec, but just "simulate" unsupported Ruby version?

Comment (deivid-rodriguez) [reply spec/other/compatibility_guard_spec.rb]: The thing is... this spec will never happen in real life I think. Running `gem install bundler` on an unsupported ruby would be short-circuit first by rubygems, wouldn't it?

Comment (deivid-rodriguez) [reply spec/other/compatibility_guard_spec.rb]: In other words, this situation will only happen if bundler 2 is installed on rubies below 2.3 and that is... impossible I think?

Comment (simi) [reply spec/other/compatibility_guard_spec.rb]: I think it is still possible (in theory, not tested) with some old ruby version and old rubygems version not supporting Ruby version constraint. But you're right. It is probably not worth it to keep this spec.

Comment (deivid-rodriguez) [reply spec/other/compatibility_guard_spec.rb]: Yeah, that's a good point. Maybe it's worth keeping actually.It's just three lines of lib code and this spec.

**Comment by shushugah** [comment]
Why are VCR files enclosed in this PR?

**Comment by deivid-rodriguez** [comment]
@shushugah I could've removed the realworld specs that were specific to old rubies, but I ended up adding them to the current suite instead. The VCR files are the result of the API responses recorded by those tests.

I did this because I thought maybe it was worth running these high level specs in our test suite (even if they test cases that only been problematic on old rubies). But I'm not really sure, I guess we could just remove them.

File: spec/other/compatibility_guard_spec.rb (hunk 1 above)
Comment (deivid-rodriguez) [reply spec/other/compatibility_guard_spec.rb]: Let's not get too agressive with the removing, I restored the compatibility guard for now :+1:.

**Comment by deivid-rodriguez** [comment]
Finally went ahead and removed that too, so no VCR files included in this PR anymore.

File: spec/other/compatibility_guard_spec.rb (hunk 1 above)
Comment (segiddins) [reply spec/other/compatibility_guard_spec.rb]: You can also clone and run Bundler locally, so I think at the bare minimum we should keep the check in lib/

**Comment by deivid-rodriguez** [comment]
@bundlerbot r=colby-swandale

**Pull request merged**